### ISSUE TITLE
GBA ROM Prefetch emulation.

### ISSUE THREAD

**Issue by ADormant** [shonumi/gbe-plus#19]
I don't know if you know but some time ago MGBA emulated apparently undocumented GBA Rom prefetch function.
http://board.byuu.org/phpbb3/viewtopic.php?f=8&t=530

Test roms
https://github.com/Gekkio/mooneye-gb#mooneye-gb-acceptance-tests

**Comment by shonumi** [comment]
Yeah, I've been aware of the work endrift and byuu have been making on the GBA ROM prefetch. From what I understand, it's mostly understood, but there's that last little percent missing to make it perfect. I'm waiting until they finish things up completely before I tackle it. And before then I need to work on GBA timing in general.

I've also spoken with gekkio on occasion. Similarly, I'm waiting for his research to end before I go and implement the things he's investigating. I'll get to it eventually.

Keep in mind complete hardware accuracy is a long-term sub-goal. It's not priority-one material, though I definitely care about it. It's just going to have to take a backseat to other things (general game compatibility, CGFX, other misc features). Once my plate clears, then I'll devote more time to this area for sure, but it'll be a while.

**Comment by ADormant** [comment]
@shonumi 
More documentation 
http://twvideo01.ubm-us.net/o1/vault/gdc07/slides/S3727i1.pdf
http://www.prodigygames.8k.com/articles/gbadftgu2.html
http://www.coranac.com/tonc/text/toc.htm
http://patater.com/gbaguy/gbaasm.htm
http://www.loirak.com/gameboy/gbatutor.php
http://www.drunkencoders.com/tutorials/GBA/tutorial_4.html
http://emu-docs.org/Game%20Boy%20Advance/AgentQ_GBASpec.txt
https://www.reinterpretcast.com/writing-a-game-boy-advance-game
https://www.csh.rit.edu/~moffitt/books/HAM_Tutorial_PDF.pdf
https://gbatemp.net/threads/nds-rom-file-format.202880/
https://gbatemp.net/threads/how-do-you-literally-open-nds-files.318242/
https://gbatemp.net/threads/nds-file-format.41783/
https://gbatemp.net/threads/gbc-hacking.388704/
http://www.romhacking.net/?page=documents&category=&platform=10&game=&author=&perpage=20&level=&title=&desc=&docsearch=Go
http://www.romhacking.net/?page=documents&category=&platform=23&game=&author=&perpage=20&level=&title=&desc=&docsearch=Go

**Comment by ADormant** [comment]
http://gekkio.fi/files/mooneye-gb/nightly/logic-analysis/external-bus/
